src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -0,0 +1,294 @@
+            cluster_environment=cluster_environment,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+
+        self.use_chunk = use_chunk
+        self.chunk_size = chunk_size
+        self.enable_distributed_storage = enable_distributed_storage
+        self.placement_policy = placement_policy
+        self.force_outputs_fp32 = force_outputs_fp32
+        self.gpu_margin_mem_ratio = gpu_margin_mem_ratio
+        self.chunk_size_search_kwargs = {
+            "search_range": chunk_search_range,
+            "n_grids": chunk_search_n_grids,
+            "min_chunk_size": min_chunk_size,
+        }
+        self.amp_kwargs = {
+            "initial_scale": initial_scale,
+            "min_scale": min_scale,
+            "growth_factor": growth_factor,
+            "backoff_factor": backoff_factor,
+            "growth_interval": growth_interval,
+            "hysteresis": hysteresis,
+            "max_scale": max_scale,
+        }
+        self._num_nodes = 1
+        self._logger = get_dist_logger()
+
+    def setup_distributed(self):
+        if not gpc.is_initialized(ParallelMode.GLOBAL):
+            disable_existing_loggers()
+            gpc.init_global_dist(
+                rank=self.global_rank,
+                world_size=self.world_size,
+                backend="nccl",
+                host=self.cluster_environment.main_address,
+                port=self.cluster_environment.main_port,
+            )
+            gpc.set_device(self.local_rank)
+        self.process_group = ProcessGroup()
+
+    def model_sharded_context(self):
+        """Provide hook to create modules in a distributed aware context. This is useful for when we'd like to
+        shard the model instantly, which is useful for extremely large models which can save memory and
+        initialization time.
+
+        Returns: Model parallel context.
+        """
+
+        class ModelShardedContext(ColoInitContext):
+            def _post_init_method(self, module: torch.nn.Module, *args, **kwargs):
+                if getattr(module, "_colossalai_module", False) is True:
+                    return
+                super()._post_init_method(module, *args, **kwargs)
+                module._colossalai_module = True
+
+        return ModelShardedContext()
+
+    def setup_precision_plugin(self) -> None:
+        super().setup_precision_plugin()
+        is_training = self.lightning_module.trainer and self.lightning_module.trainer.training
+        if is_training:
+            if len(self.optimizers) > 1:
+                raise MisconfigurationException("ColossalAIStrategy only supports single Optimizer now.")
+            optimizer = self.optimizers[0]
+            assert isinstance(optimizer, (CPUAdam, HybridAdam)), (
+                "ColossalAIStrategy only supports colossalai.nn.optimizer.CPUAdam "
+                "and colossalai.nn.optimizer.HybridAdam."
+            )
```

Comment:
Let's make this a regular runtime error, not an assertion. An assertion must always hold true, but here we are dealing with user input.

Reply:
I've changed it.